lib/api.es6.js

```diff
@@ -50,6 +47,102 @@ const login = (apiOptions, username, pass) => {
   });
 };
 
+const register = (apiOptions, username, password, email, newsletter) => {
+  return new Promise((r, x) => {
+    if (!apiOptions.oauthAppOrigin) {
+      x('Please set up a Reddit Oauth App, and pass in its URL as oauthAppOrigin to config.');
+    }
+
+    if (!apiOptions.clientId) {
+      x('Please set up a Reddit Oauth App, and pass in its id as clientId to config.');
+    }
+
+    if (!apiOptions.clientSecret) {
+      x('Please set up a Reddit Oauth App, and pass in its secret as clientSecret to config.');
+    }
+
+    const data = {
+      user: username,
+      passwd: password,
+      passwd2: password,
+      api_type: 'json',
+    };
+
+    if (email) {
+      data.email = email;
+    }
+
+    if (newsletter) {
+      data.newsletter_subscribe = true;
+
+      if (!email) {
+        x('NEWSLETTER_NO_EMAIL');
+      }
+    }
+
+    const endpoint = `${apiOptions.origin}/api/register`;
+    const s = btoa(`${apiOptions.clientId}:${apiOptions.clientSecret}`);
+    const basicAuth = `Basic ${s}`;
+    const headers = {
+      'User-Agent': apiOptions.userAgent,
+      'Authorization': basicAuth,
+      ...apiOptions.defaultHeaders,
+    };
+
+    try {
+      const timeout = 20000;
+      superagent
+        .post(endpoint)
+        .set(headers)
+        .type('form')
+        .send(data)
+        .timeout(10000)
+        .end((err, res) => {
+          const obj = {};
+          if (err || !res.ok) {
+            obj.status = err.timeout ? 504 : res.status || 500;
+            obj.errorType = obj.status;
+          }
+
+          const errors = parseError(res);
+          if (errors.length) {
+            return x(errors);
+          }
+
+          const cookies = parseCookies(res);
+
+          const redditSessionExists = sessionExists(res);
+          if (redditSessionExists) {
+            return convertCookiesToAuthToken(apiOptions, cookies).then(r,x);
+          }
+          x('UNKNOWN_ERROR');
+       });
+    } catch (err) {
+      return x('Invalid registration information.');
+    }
+  });
+}
+
+// Parses the error message for login and register functions
+const parseError = (res) => {
+  if (res.body.json && res.body.json.errors && res.body.json.errors.length) {
+    return res.body.json.errors;
+  }
+  return [];
+}
+
+// Parses the cookies for the login and register functions
+const parseCookies = (res) => {
+  return (res.header['set-cookie'] || []).map(c => {
+    return c.split(';')[0];
+  });
+}
+
+// Parses set-cookie header to see if the session exists
+const sessionExists = (res) => {
+  return res.header['set-cookie'].join('').indexOf('reddit_session');
```

Comment:
to fix the bug phil pointed out above, you can use the `~` operator with `!`:
```es6
return !!~res.header['set-cookie'].join('').indexOf('reddit_session');
```

or, since that looks a bit ridiculous :), you can just use a `> -1` check;